lib/helpers/model/applyHooks.js

```diff
@@ -164,3 +150,33 @@ function applyHooks(model, schema, options) {
       createWrapper(method, originalMethod, null, customMethodOptions);
   }
 }
+
+/**
+ * Check if there is an embedded schematype in the given schematype. Handles drilling down into primitive
+ * arrays and maps in case of array of array of subdocs or map of subdocs.
+ *
+ * @param {SchemaType} curType
+ * @returns { childModel: Model | typeof Subdocument, curType: SchemaType } | null
```

Comment:
```suggestion
 * @returns {{ childModel: Model | typeof Subdocument, curType: SchemaType } | null}
```

I think this is more correct JSDoc, as otherwise `| null` is seen as the description.